In [1]:
import numpy as np
import pandas as pd
from scipy.stats import mode
import warnings
warnings.filterwarnings("ignore")

Q1. What are the three measures of central tendency?

Ans:- Mean, Median, Mode is three central tendency

Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?

Ans:- Mean is average of data, median is the middle value of sorted data and mode is most frequent value.

Q3. Measure the three measures of central tendency for the given height data:

 [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [2]:
height =  [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
height = np.array(height)

In [3]:
print(f'mean is {np.mean(height)}, median is is {np.median(height)}and mode is {mode(height).mode}')


mean is 177.01875, median is is 177.0and mode is [177.]


In [4]:
# Q4. Find the standard deviation for the given data:

data = np.array([178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5])
np.std(data)

1.7885814036548633

Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe
the spread of a dataset? Provide an example.

In [5]:
data = np.array([1,2,3,4,5,6,7,8,9])

# RNAGE:- here range tells that the difference between the max value and min value, if the difference is less then the spread is less and values are concentrated. So range is 9-1 = 8

# VARIANCE:- it is average of sum of square of deviation of each value from the mean, lesser the values are separated from the mean, concentrated the values are.
np.var(data)

6.666666666666667

STANDARD DEVIATION:- It is sqrt of variance, as it is not the square term of the errors like variance, it provide and understandable value which is comparable to mean.


In [6]:
np.std(data)

2.581988897471611

Q6. What is a Venn diagram?

Ans:- A Venn diagram is a graphical representation of the relationships between sets. It is a diagram that uses circles or other shapes to represent sets and their relationships to one another. The size and position of the circles in the diagram indicate the degree of overlap or inclusion of the sets.

In [7]:
# Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:

# (i) 	A B

# (ii)	A ⋃ B
A = {2,3,4,5,6,7}
B = {0,2,6,8,10}
print(f'intersection of A and B is {A & B}')
print(f'union of A and B is {A | B}')

intersection of A and B is {2, 6}
union of A and B is {0, 2, 3, 4, 5, 6, 7, 8, 10}


Q8. What do you understand about skewness in data?

Ans:- When we plot the data and find that the distribution is not normally distributed but one side is extended called skewed data.
example:- suppose we plot the graph of salary of employee, then we found that mode is 30,000/- but few managers are earning in 5-6 lakh per month, hence distribution will tilt towards right known as right skewed distribution, in a similar way data can be left skewed.

Q9. If a data is right skewed then what will be the position of median with respect to mean?

Ans:- If data is right skewed then Mean >Median >Mode

Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?
Ans:- Covariance tells the relation between two variables , higher the value of covariance more they are increasing or decreasing with each other but the value can lies in any range, hence we cannot tell out of two covariance which set of data is more corelated. This problem is solved by cofficient of covarianc called corelation, its value lies between [-1,1] -1 shows that variables are inversely corelated, i.e. if value of first increase other will decrease, 0 value of correlation shows that there is no relation between both variables, value near to 1 shows high positive relation.

Q11. What is the formula for calculating the sample mean? Provide an example calculation for a
dataset.



In [8]:
# Ans:- mean = sum of values / no. of values
data = [1,2,3,4]
mean = (1+2+3+4)/4
mean

2.5

In [9]:
Q12. For a normal distribution data what is the relationship between its measure of central tendency?

Ans:- Mean = median= Mode

SyntaxError: invalid syntax (79700234.py, line 3)

Q13. How is covariance different from correlation?
Ans:- Covariance tells the relation between two variables , higher the value of covariance more they are increasing or decreasing with each other but the value can lies in any range, hence we cannot tell out of two covariance which set of data is more corelated. This problem is solved by cofficient of covarianc called corelation, its value lies between [-1,1] -1 shows that variables are inversely corelated, i.e. if value of first increase other will decrease, 0 value of correlation shows that there is no relation between both variables, value near to 1 shows high positive relation.

In [ ]:
# Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.
height = [1,2,3,3,4] 
print(f'mean is {np.mean(height)}, median is{np.median(height)}, mode is {mode(height).mode} and standard deviation is {np.std(np.array(height))}')
print('-'*100)
height = [1,2,3,3,4, 1000] 
print(f'mean is {np.mean(height)}, median is{np.median(height)} and mode is {mode(height).mode}and standard deviation is {np.std(np.array(height))}')
height = [1,2,3,3,4, 1000] 

Ans:- when outliers enters in a data set then measures of central tendency and dispersion get effected.
before outliear values are mean is 2.6, median is3.0and mode is 3 and range is 3, but after outlier values are mean is 168.83333333333334, median is3.0and mode is 3 and range become 999, shows that mean and range effect very high while mean and mode least, standard deviation also changes from 1 to 371.